examples/motoko_examples/basic-dao/src/basic_dao.ts

```diff
@@ -0,0 +1,321 @@
+import {
+  Query,
+  Update,
+  int32,
+  Principal,
+  Result,
+  ic,
+  Opt,
+  UpdateAsync,
+} from 'azle';
+import * as Types from './types';
+
+// we should get the init state when we install the canister
+/* stable */
+let init: Types.BasicDaoStableStorage = {
+  accounts: [],
+  proposals: [],
+  system_params: {
+    transfer_fee: { amount_e8s: 0 },
+    proposal_vote_threshold: { amount_e8s: 0 },
+    proposal_submission_deposit: { amount_e8s: 0 },
+  },
+};
+
+/* stable */ let installed = false;
+/* stable */ let installer: string | null = null;
+/* stable */ let accounts = Types.accounts_fromArray(init.accounts);
+/* stable */ let proposals = Types.proposals_fromArray(init.proposals);
+/* stable */ let next_proposal_id: int32 = 0;
+/* stable */ let system_params: Types.SystemParams = init.system_params;
+
+// function heartbeat() {
+//   await execute_accepted_proposals();
+// };
+
+// simulate init function
+// for now as a hack we need to call this function manually
+export function initialize(init: Types.BasicDaoStableStorage) {
+  if (installed) throw new Error('Already installed');
+  installed = true;
+  installer = ic.caller();
+  init = init;
+}
+
+function account_get(id: Principal): Opt<Types.Tokens> {
+  return accounts.get(id) ?? null;
+}
+
+function account_put(id: Principal, tokens: Types.Tokens) {
+  accounts.set(id, tokens);
+}
+
+function proposal_get(id: int32): Opt<Types.Proposal> {
+  return proposals.get(id) ?? null;
+}
+
+function proposal_put(id: int32, proposal: Types.Proposal) {
+  proposals.set(id, proposal);
+}
+
+/// Transfer tokens from the caller's account to another account
+export function* transfer(
+  transfer: Types.TransferArgs
+): UpdateAsync<Result<void, string>> {
+  let caller = ic.caller();
+  let from_tokens = account_get(caller);
+  switch (from_tokens) {
+    case null: {
+      return yield { err: 'Caller needs an account to transfer funds' };
+    }
+    case from_tokens as Types.Tokens: {
```

Comment:
switch doesn't play nicely with Opt, since we need casting here, is there a better way?